snap/snapshots.go

```diff
@@ -27,27 +27,81 @@ import (
 	"path/filepath"
 	"strings"
 
-	"gopkg.in/yaml.v2"
-
 	"github.com/snapcore/snapd/strutil"
+	"gopkg.in/yaml.v2"
 )
 
 var osOpen = os.Open
 
-// SnapshotOptions contains the information provided by a snap package about
-// what files should be included in the snapshots.
+// SnapshotOptions describes the options available for snapshots.
+// The initial source of these options is a file in the snap package.
+// In addition, options can be modified with dynamic requests via REST API.
 type SnapshotOptions struct {
-	// ExcludePaths is the list of file and directory patterns that need to be
+	// Exclude is the list of file and directory patterns that need to be
 	// excluded from a snapshot. At the moment the only supported globbing
 	// character is "*", which stands for any sequence of characters other than
 	// "/".
-	ExcludePaths []string `yaml:"exclude"`
+	Exclude []string `yaml:"exclude" json:"exclude,omitempty"`
 }
 
 const (
 	snapshotManifestPath = "meta/snapshots.yaml"
 )
 
+// IsEmpty determines if SnapshotOptions structure is empty.
+//
+// For this function "empty" is defined as only containing empty containers
+// with no values. The purpose is to determine if representing the structure
+// in JSON would provide more than just keys, braces, brackets.
+func (opts *SnapshotOptions) IsEmpty() bool {
+	return len(opts.Exclude) == 0
+}
+
+// Merge combines existing with additional options.
+func (opts *SnapshotOptions) Merge(moreOptions *SnapshotOptions) error {
+	if moreOptions == nil {
+		return nil
+	}
+	if err := moreOptions.Validate(); err != nil {
```

Comment:
I'd have expected the `moreOptions` to be pre-validated going into `Merge`, it seems like this would be out of the responsibilites of `Merge`

Reply:
Implemented dry run concept to pre-validate as per discussion.